### PR TITLE
Fulfil booking requests

### PR DESCRIPTION
**Pull request by benlovell**
Allows the booking manager to fulfil booking requests. This can be reviewed on a commit-by-commit basis but I'll be squashing it down to something more meaningful for permanent record.

### CODE REVIEW

File: app/forms/appointment_form.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  end
+
+  def date
+    @date ||= location_aware_booking_request.primary_slot.date
+  end
+
+  def time
+    @time ||= location_aware_booking_request.primary_slot.delimited_from
+
+    Time.zone.parse("2016-01-01 #{@time}")
```
Comment (dwhenry) [inline app/forms/appointment_form.rb]: given the date part is ignored can we not just use 
```
 Time.zone.parse("#{@time}")
```
or does it break with daylight saving?

Comment (benlovell) [inline app/forms/appointment_form.rb]: Nope, that just never occurred to me. I'll change it 👍

File: app/models/slot.rb [hunk 2]
```diff
@@ -15,11 +15,19 @@ def morning?
   end
 
   def to_s
-    "#{date.to_s(:gov_uk)} - #{period}"
+    "#{formatted_date} - #{period}"
+  end
+
+  def formatted_date
+    date.to_s(:gov_uk)
```
Comment (benlovell) [inline app/models/slot.rb]: This should be a `govuk_date` instead.

**Comment by benlovell** [comment]
@benbarnett Lovely job.

**Comment by benlovell** [comment]
Squashed that fixup commit.

**Pull request merged**